lib/constructs/interface.js

```diff
@@ -64,9 +93,12 @@ Interface.prototype.generateConstructor = function () {
 }\n`;
   }
 
-  if (this.idl.inheritance) {
+  if (this.idl.inheritance && !utils.isGlobal(this.idl)) {
     this.str += `${this.name}.prototype = Object.create(${this.idl.inheritance}.interface.prototype);
 ${this.name}.prototype.constructor = ${this.name};\n`;
+  } else if (utils.isGlobal(this.idl)) {
+    this.str += `${this.name}.prototype = Object.create(${this.name}Properties);
```

Comment:
Do `Object.setPrototypeOf(obj, namedPropertiesObject)`. Actually the above inheritance should probably do that too (although in that case you'll also need `Object.setPrototypeOf(Class, SuperClass)`.